client/lib/cgutil/cpuset_manager_v2.go

```diff
@@ -80,6 +86,26 @@ func NewCpusetManagerV2(parent string, reservable []uint16, logger hclog.Logger)
 	}
 }
 
+// minimumControllers sets the minimum set of required controllers on the
+// /sys/fs/cgroup/cgroup.subtree_control file - ensuring [cpuset, cpu, io, memory, pids]
+// are enabled.
+func minimumRootControllers() error {
+	e := new(editor)
+	s, err := e.read("cgroup.subtree_control")
+	if err != nil {
+		return err
+	}
+	required := set.From[string]([]string{"cpuset", "cpu", "io", "memory", "pids"})
+	enabled := set.From[string](strings.Fields(s))
+	needed := required.Difference(enabled)
+	sb := new(strings.Builder)
+	for _, controller := range needed.List() {
+		sb.WriteString("+" + controller + " ")
```

Comment:
Is having a trailing ` ` ok or do we need to trim whitespace when we're done?

Reply:
experimentally it's safe but yeah I can clean this up. can also avoid writing an empty string in the case there is nothing to write